### ISSUE TITLE
Tree: Component should be skippable by tabbing

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5334]
Testing the tree component I just realized that we cannot skip the component and apply roving tabindex, so enter in a tree can be a nightmare and it could need thousen of tab pressings to get to the next focuseable element in the page
https://storybook.clayui.com/?path=/story/design-system-components-treeview--multiple-selection

https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_roving_tabindex
https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/#keyboard-interaction-25

After review ARIA patterns it is my **Proposal**:
- Moves focus to the next interactive widget in the current element (tree node).
- If there are no more interactive controls in the current element, moves focus out of the tree.

**Comment by matuzalemsteles** [comment]
We were not able to fully apply the ARIA attributes of treegrid completely to the TreeView because we have an explicit tree instead of a list as is the case with the treegrid but I think this is the way to go but this path is still complex and requires more work first we need to work on #5030 that will allow to create the virtualization of the tree that consequently turns it into a list and to get accessibility we will apply the treegrid attributes. Initially research was done and an attempt was made to apply this pattern, but without much success, you can read the PR comments https://github.com/liferay/clay/pull/5045#issuecomment-1249980691

What we can follow for now is to use the `useNavigation` hook and the focus system for now which will implement the navigation shortcut keys not completely but the main ones.

Basically our component follows the accessibility rules of https://www.w3.org/WAI/ARIA/apg/patterns/treeview/examples/treeview-1a/, so we are still compliance.
